src/helpers/afmHelper.ts

```diff
@@ -39,16 +39,21 @@ export function getMasterMeasureObjQualifier(afm: AFM.IAfm, localIdentifier: AFM
     return null;
 }
 
-const getDateFilter = (filters: AFM.FilterItem[]): AFM.DateFilterItem => {
+const getDateFilter = (filters: AFM.ExtendedFilter[]): AFM.DateFilterItem => {
     return filters.find(AfmUtils.isDateFilter);
 };
 
-const getAttributeFilters = (filters: AFM.FilterItem[] = []): AFM.AttributeFilterItem[] => {
+const getAttributeFilters = (filters: AFM.ExtendedFilter[] = []): AFM.AttributeFilterItem[] => {
     return filters.filter(AfmUtils.isAttributeFilter).filter(negate(AfmUtils.isAttributeFilterSelectAll));
 };
 
-export const mergeFiltersToAfm = (afm: AFM.IAfm, additionalFilters: AFM.FilterItem[]): AFM.IAfm => {
+const getMeasureValueFilters = (filters: AFM.ExtendedFilter[] = []): AFM.IMeasureValueFilter[] => {
+    return filters.filter(AFM.isMeasureValueFilter);
+};
+
+export const mergeFiltersToAfm = (afm: AFM.IAfm, additionalFilters: AFM.ExtendedFilter[]): AFM.IAfm => {
     const attributeFilters = getAttributeFilters(additionalFilters);
     const dateFilter = getDateFilter(additionalFilters);
-    return AfmUtils.appendFilters(afm, attributeFilters, dateFilter);
+    const measureValueFilters = getMeasureValueFilters(additionalFilters);
```

Comment:
Is the purpose of the code around just to remove the `All` filter? If it is so, it can be simplified made clearer.

Reply:
I don't think so. For date filters, it picks just the first one of collection and merges it into AFM, omitting the others. For attribute filters, it optimizes them (possibly to work around some logic in AD that is not ideal, I'm not sure). There is no reason why to touch measure value filters but it needs to be there, otherwise, they would be dropped. 

I would leave it as it is or do you have a better solution in mind?